File: src/types/order.ts
export interface TaxBreakdown {
  state_rate: number;
  county_rate: number;
  city_rate: number;
  special_rate: number;
}

export interface BackendOrder {
  id: number;
  latitude: number;
  longitude: number;
  composite_tax_rate: number;
  tax_amount: number;
  total_amount: number;
  breakdown: TaxBreakdown;
  jurisdictions: string[];
  status: string;
  reporting_code: string;
  created_at: string;
  updated_at: string;
}

export interface Order {
  id: number;
  latitude: number;
  longitude: number;
  composite_tax_rate: number;
  tax_amount: number;
  total_amount: number;
  breakdown: TaxBreakdown;
  jurisdictions: string[];
  status: string;
  reporting_code: string;
  created_at: string;
  updated_at: string;
}

export interface CreateOrderPayload {
  latitude: number;
  longitude: number;
  subtotal: number;
  timestamp?: string;
}

export interface OrderFilters {
  dateFrom?: string;
  dateTo?: string;
  amountMin?: number;
  amountMax?: number;
  sortBy?: string;
  sortDir?: "asc" | "desc";
  status?: string;
  reportingCode?: string;
}

export interface PaginatedMeta {
  page: number;
  perPage: number;
  total: number;
  totalPages: number;
}

export interface PaginatedResponse<T> {
  data: T[];
  meta: PaginatedMeta;
}

File: internal/entity/order.go
package entity

import "time"

// could use decimal.Decimal for more precision
type OrderStatus string
type Order struct {
	Id        int     `json:"order"`
	Latitude  float64 `json:"latitude"`
	Longitude float64 `json:"longitude"`

	TotalAmount float64 `json:"total_amount"`
	TaxAmount   float64 `json:"tax_amount"`

	CompositeTaxRate float64          `json:"composite_tax_rate"`
	Breakdown        TaxRateBreakdown `json:"breakdown"`

	Jurisdictions []string `json:"jurisdictions"`
	ReportingCode string   `json:"reporting_code"`

	Status OrderStatus

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type TaxRateBreakdown struct {
	StateRate   float64 `json:"state_rate"`
	CountyRate  float64 `json:"county_rate"`
	CityRate    float64 `json:"city_rate"`
	SpecialRate float64 `json:"special_rate"`
}
